pkg/oauth/user.go

```diff
@@ -207,6 +210,64 @@ func (s *server) Login(c echo.Context) error {
 	return c.NoContent(http.StatusNoContent)
 }
 
+var (
+	errInvalidLogoutRedirectURI = errors.DefineInvalidArgument(
+		"invalid_logout_redirect_uri",
+		"the redirect URI did not match the one(s) defined in the client",
+	)
+	errMissingAccessTokenIDParam = errors.DefinePermissionDenied(
+		"missing_param_access_token_id",
+		"access token ID was not provided",
+	)
+)
+
+func (s *server) ClientLogout(c echo.Context) error {
+	ctx := c.Request().Context()
+	accessTokenID := c.QueryParam("access_token_id")
+	var redirectURI string
+	if accessTokenID == "" {
+		return errMissingAccessTokenIDParam
+	}
+	at, err := s.store.GetAccessToken(ctx, accessTokenID)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+	}
+	if at != nil {
+		client, err := s.store.GetClient(ctx, &at.ClientIDs, &types.FieldMask{Paths: []string{"logout_redirect_uris"}})
+		if err != nil {
+			return err
+		}
+		if err = s.store.DeleteAccessToken(ctx, accessTokenID); err != nil {
+			return err
+		}
+		s.removeAuthCookie(c)
```

Comment:
We don't know that the auth cookie corresponds with the session we're deleting. If it doesn't, we may leave a session active while the user thinks it's revoked.

Reply:
I can't think of any realistic scenario where that would be the case. The user has obtained the access token using this auth cookie (session). Since our official clients tie them together, there should be no corruptions or mixups. So I think this is unelectable. In any case we still have the old logout handler, so clients can decide for themselves.

Reply:
As discussed, the logout will now
- delete the access token
- delete the session that was associated with the access token
- delete the session associated with the current auth cookie of the user (if it differs from the sessions stored in the access token)